dependencies.yaml

```diff
@@ -406,6 +407,26 @@ dependencies:
               - *libcusolver114
               - *libcusparse_dev114
               - *libcusparse114
+      - output_types: pyproject
+        matrices:
+          - matrix:
+              cuda: "12.*"
+            packages:
+              - nvidia-cublas-cu12
+              - nvidia-cufft-cu12
+              - nvidia-curand-cu12
+              - nvidia-cusparse-cu12
+              - nvidia-cusolver-cu12
```

Comment:
Are we missing nvJitLink in this list? It is listed in CMakeLists.txt and it is excluded in auditwheel.
```suggestion
              - nvidia-cusolver-cu12
              - nvidia-nvjitlink-cu12
```

Reply:
I don't think nvjitlink is a direct dependency. It's excluded by auditwheel because auditwheel traverses the entire dependency tree, not just the direct dependencies. I'm not sure if we need it in the rpath or not.

Reply:
You are correct, nvjitlink was added as a dependency of one of these libraries (I forget which one) in a relatively recent CUDA version (maybe 12.5). Whichever library it is should have its RPATH set appropriately to find nvjitlink if it needs it.

Reply:
Looks like we need `nvjitlink` in the rpath:

https://github.com/rapidsai/cuml/actions/runs/10406862210/job/28821649611?pr=5966#step:8:475